src/assignments/tech-jobs-oo.rst

```diff
@@ -1,6 +1,586 @@
+to the ``Employer`` class for hints on how to structure your code.
+
+The ``Location`` Class
+^^^^^^^^^^^^^^^^^^^^^^^
+
+Open the ``Location.java`` file. Note that the getters, setters, and custom
+methods for this class are done, as is the constructor for initializing the
+``id`` field.
+
+Sally left you a ``TODO`` comment with instructions for coding a second
+constructor:
+
+#. It should call the first constructor to initialize the ``id`` field.
+#. It must also initialize the ``value`` field for a new ``Location`` object.
+
+The ``CoreCompetency`` Class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open the class file. In this case, the constructors and custom methods are
+ready. Sally needs you to complete the somewhat tedious task of writing the
+getters and setters for the ``id`` and ``value`` fields, but NOT for
+``nextID``.
+
+Fortunately, IntelliJ has a tool to help with this:
+
+#. Right-click in the editor pane and select *Generate*.
+#. Select the *Getter and Setter* option.
+#. Select the ``id`` and ``value`` options, then click *OK*.
+
+PRESTO! Getters and setters appear.
+
+The ``PositionType`` Class
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Open the class file. This time the constructors, getters, and setters are done.
+Sally's comments direct you to where you need to add the custom methods.
+
+#. Code a ``toString`` method that just returns the ``value`` of a
+   ``PositionType`` object.
+#. Use the *Generate* option again to add the ``equals`` and ``hashCode``
+   methods. Refer to the :ref:`previous section <equals-shortcut>` of this
+   chapter if you need a quick review.
+#. Assume that two ``PositionType`` objects are equal when their ``id`` fields
+   match.
+
+.. admonition:: Tip
+
+   Now would be a good time to save, commit, and push your work up to GitHub.
+
+3) Complete the ``Job`` Class
+------------------------------
+
+Now open the ``Job`` file. OOF! There are a lot of fields declared and not much
+else.
+
+#. Code a constructor to initialize the ``id`` field with a unique value. This
+   constructor should take no parameters.
+#. Code a second constructor that takes 5 parameters and assigns values to
+   ``name``, ``employer``, ``location``, ``positionType``, and
+   ``coreCompetency``. Also, this constructor should call the first in order to
+   initialize the ``id`` field.
+#. Generate getters and setters for each field EXCEPT ``nextID``.
+#. Generate the ``equals`` and ``hashCode`` methods. Consider two ``Job``
+   objects equal when their id fields match.
+
+.. admonition:: Tip
+
+   Save, commit, and push your work to GitHub.
+
+4) Use Unit Testing to Verify Parts of the ``Job`` Class
+---------------------------------------------------------
+
+Instead of manually creating sample ``Job`` objects to verify that your class
+works correctly, you will use unit tests instead.
+
+Create a new package inside the ``TechJobsOO`` folder called ``Tests``, then
+create a new class inside this folder called ``JobClassTests``. The file will
```

Comment:
While not a strict convention, the test class for a `Job` class would almost always be named `JobTest`.

Reply:
OK, fixed.  However, you will need to explain to me why we must use the singular ``JobTest`` when the file holds multiple tests of the ``Job`` class.